packages/theme-predictive-search/src/request.js

```diff
@@ -0,0 +1,82 @@
+export default function request(configParams, query, onSuccess, onError) {
+  var xhr = new XMLHttpRequest();
+
+  xhr.onreadystatechange = function() {
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      var contentType = xhr.getResponseHeader("Content-Type");
+
+      if (xhr.status >= 500) {
+        onError(new Error("Server Error"));
+
+        return;
+      }
+
+      if (
+        typeof contentType !== "string" ||
+        contentType.toLowerCase().match("application/json") === null
+      ) {
+        onError(new Error("Request Error"));
+
+        return;
+      }
+
+      if (xhr.status === 200) {
+        try {
+          var res = JSON.parse(xhr.responseText);
+          res.query = query;
+          onSuccess(res);
+        } catch (error) {
+          onError(error);
+        }
+
+        return;
+      }
+
+      if (xhr.status === 429) {
+        try {
+          var throttledJson = JSON.parse(xhr.responseText);
+          var throttledError = new Error();
+
+          throttledError.name = throttledJson.message;
+          throttledError.message = throttledJson.description;
+          throttledError.retryAfter = xhr.getResponseHeader("Retry-After");
+
+          onError(throttledError);
+        } catch (error) {
+          onError(error);
+        }
+
+        return;
+      }
+
+      if (xhr.status) {
```

Comment:
Not sure if we need the `if` here => would we ever have a point when `xhr.status` is not truthy?
If we do, then what happens in that case?
I think this is fine, I'm just not 100% sure.